modules/on_gke/main.tf

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//*****************************************
+//  Create Keys for Forseti server and
+//  client service accounts.
+//*****************************************
+
+resource "google_service_account_key" "server_key" {
+  service_account_id = "${var.forseti_server_service_account}"
+}
+
+resource "google_service_account_key" "client_key" {
+  service_account_id = "${var.forseti_client_service_account}"
+}
+
+//*****************************************
+//  Create Keys for Forseti server and
+//  client service accounts.
+//*****************************************
+
+resource "google_project_iam_member" "cluster_service_account-storage_reader" {
+  project = "${var.project_id}"
+  role    = "roles/storage.objectViewer"
+  member  = "serviceAccount:${var.gke_service_account}"
```

Comment:
Is the `gke_service_account` supposed to be created manually by the user beforehand?

Reply:
Not necessarily.  The best practice is to create a new IAM service account for a GKE cluster.  In other words, don't use the default compute service account.  This is created via the [terraform-google-kubernetes-engine](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine) module

In the examples/on_gke/new_gke_cluster example, this is an output from the terraform-google-kubernetes-engine module.

https://github.com/kevensen/terraform-google-forseti/blob/1c738720e2ee373175ea8f92deb50d44ffab9112/examples/on_gke/new_gke_cluster/main.tf#L154

Reply:
Ok thanks for the clarification!